packages/devtools-core/src/core/inspect/component-handle/class.ts

```diff
@@ -0,0 +1,193 @@
+/*!
+ * V4Fire DevTools
+ * https://github.com/V4Fire/DevTools
+ *
+ * Released under the MIT license
+ * https://github.com/V4Fire/DevTools/blob/main/LICENSE
+ */
+import SyncPromise from 'core/promise/sync';
+import { ComponentQuery, deserialize } from '@v4fire/devtools-backend';
+
+// FIXME: This import confronts with the V4Fire restrictions @see https://github.com/V4Fire/Client/issues/1111
+import type iBlock from 'components/super/i-block/i-block';
+import type { ComponentElement } from 'core/component';
+
+import DevtoolsHandle, { PageFunctionWithCtx, evaluateEngine } from 'core/inspect/devtools-handle';
+import type { ComponentData } from 'features/components/b-components-panel/interface';
+
+export default class ComponentHandle extends DevtoolsHandle<iBlock> {
+	/**
+	 * Component query
+	 */
+	private readonly query: ComponentQuery;
+
+	/**
+	 * Promise of component handle readiness
+	 */
+	private isReadyStore?: SyncPromise<void>;
+
+	constructor(query: ComponentQuery) {
+		const id = `component-${query.componentId}`;
+		super(id);
+		this.query = query;
+	}
+
+	/**
+	 * Returns the component id
+	 */
+	get componentId(): string {
+		return this.query.componentId;
+	}
+
+	/**
+	 * Returns the component name
+	 */
+	get componentName(): string | undefined {
+		return this.query.componentName;
+	}
+
+	override evaluate<T = unknown, Arg = unknown>(
+		pageFunction: PageFunctionWithCtx<iBlock, T, Arg>,
+		arg?: Arg | undefined
+	): Promise<T> {
+		return this.isReady.then(super.evaluate.bind(this, pageFunction, arg));
+	}
+
+	/**
+	 * Highlights the component
+	 * @param autoHide
+	 */
+	highlight(autoHide: boolean): void {
+		this.evaluate((ctx, autoHide) => {
+			const backend = globalThis.__V4FIRE_DEVTOOLS_BACKEND__;
+			const node = backend.findComponentNode(ctx);
+			// @ts-expect-error Non-standard API
+			node?.scrollIntoViewIfNeeded(false);
+
+			backend.componentHighlight.show(ctx);
+
+			if (autoHide) {
+				backend.componentHighlight.hide({delay: 1500, animate: true});
+			}
+		}, autoHide).catch(stderr);
+	}
+
+	/**
+	 * Returns the data of the component
+	 */
+	async getData(): Promise<ComponentData | null> {
+		const serializedData = await this.evaluate((ctx) => {
+			const backend = globalThis.__V4FIRE_DEVTOOLS_BACKEND__;
+
+			const restricted = new Set([
+				'r',
+				'self',
+				'unsafe',
+				'router',
+				'LANG_PACKS'
+			]);
+
+			const node = backend.findComponentNode(ctx);
+
+			if (node == null) {
+				return null;
+			}
+
+			const {component} = <ComponentElement>node;
+
+			if (component == null) {
+				throw new Error('DOM node doesn\'t have component property');
```

Comment:
Может в таких кейсах делать так? Как будто эти слеши ухудшают читаемость

```suggestion
				throw new Error("DOM node doesn't have component property");
```